palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/ResourceService.java

```
@@ -15,61 +15,126 @@
+import uk.gov.gchq.palisade.service.palisade.request.GetResourcesByTypeRequest;
 import uk.gov.gchq.palisade.service.palisade.web.ResourceClient;
 
+import java.io.IOException;
+import java.io.InputStream;
 import java.net.URI;
-import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.CompletionStage;
 import java.util.concurrent.Executor;
 import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class ResourceService implements Service {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ResourceService.class);
     private final ResourceClient client;
     private final Supplier<URI> uriSupplier;
+    private final ObjectMapper objectMapper;
     private final Executor executor;
 
-    public ResourceService(final ResourceClient resourceClient, final Supplier<URI> uriSupplier, final Executor executor) {
+    private final Serialiser<LeafResource> serialiser = new LineSerialiser<>() {
+        @Override
+        public LeafResource deserialiseLine(final String line) {
+            try {
+                return objectMapper.readValue(line, AbstractLeafResource.class);
+            } catch (JsonProcessingException e) {
+                LOGGER.error("Encountered JSONProccessingException while deserialising line {}", line);
+                LOGGER.error("Exception was ", e);
+                throw new RuntimeException(e);
+            } catch (IOException e) {
+                LOGGER.error("Encountered IOException while deserialising line {}", line);
+                LOGGER.error("Exception was ", e);
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public String serialiseLine(final LeafResource obj) {
+            LOGGER.warn("No implementation of serialiseLine, ignoring argument {}", obj);
+            return null;
```

should this be returning null or an empty string?  are there non null checks for the methods that call this?

---

after some discussion with @ac74475, this should probably throw an error instead of returning null as it should never call the serialising functionality of the line serialiser